### PR TITLE
Prevent checkout of unapproved assets

### PR DESCRIPTION
**Pull request by aronhoyer**
Displays an error message if the asset the user tries to checkout
- an unapproved data product
- an asset that isn't a data product

Resolves [AB#81116](https://dev.azure.com/EquinorASA/d02c350d-092d-41bf-a812-f05d5d8ad1b0/_workitems/edit/81116)

### CODE REVIEW

File: model/Asset.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ type AssetInit = {
   id: string
   name: string
   description?: string
+  approved?: boolean
```
Comment (wenche) [inline model/Asset.ts:5]: Is this not mandatory in collibra?

File: pages/checkout/terms.tsx [hunk 2]
```diff
@@ -166,7 +168,7 @@ const CheckoutTermsView: NextPage<Props> = ({ asset, error, rightsToUse }) => {
             </>
           )
           ) }
-          {!asset && <NoAsset />}
+          {(!asset && !error) && <NoAsset />}
```
Comment (wenche) [inline pages/checkout/terms.tsx:171]: 😅

**Review by wenche — APPROVED** [review]
LGTM

File: model/Asset.ts (hunk 1 above)
Comment (aronhoyer) [reply model/Asset.ts]: the status field is, but the field will always have a value regardless of what the status is, so i made it mandatory :+1:

**Pull request merged**